File: app/src/main/java/com/example/pam_project/features/categories/edit/EditCategoryView.java
package com.example.pam_project.features.categories.edit;

import com.example.pam_project.features.categories.list.CategoryInformation;

public interface EditCategoryView {

    void bindCategory(CategoryInformation model);

    void onCategoryRetrievedError();

    void onCategoryDelete();

    void onCategoryUpdateError();

    void onCategoryDeletedError();

    void showDeleteDialog();
}

File: app/src/main/java/com/example/pam_project/features/categories/edit/EditCategoryView.kt
package com.example.pam_project.features.categories.edit

import com.example.pam_project.features.categories.list.CategoryInformation

interface EditCategoryView {
    fun bindCategory(model: CategoryInformation?)
    fun onCategoryRetrievedError()
    fun onCategoryDelete()
    fun onCategoryUpdateError()
    fun onCategoryDeletedError()
    fun showDeleteDialog()
}